src/frontends/android/app/src/main/java/org/strongswan/android/ipc/VpnProfileCrud.java

```diff
@@ -49,10 +50,15 @@ public boolean createVpnProfile(Bundle vpnProfile) {
 
     private boolean installCertificatesFromBundle(Bundle vpnProfile) {
         try {
-            createLocalKeystore();
-            String id = generateAndSetCertificateIdInBundle(vpnProfile);
-            installUserCertificateFromBundle(vpnProfile, id);
-            installCaCertificateFromBundle(vpnProfile, id);
+            if(isThereUserOrCaCertificateInBundle(vpnProfile)) {
```

Comment:
This is used in remote calls so in the end we return boolean value that is why we return it here. But in most cases I would use exceptions.